### PR TITLE
Writing Native Bears Guide: Add regex tutorial links

### LINKED ISSUE #4753: Writing-native-bears guide: Add link for short tutorial of regex

### PR DESCRIPTION
**Pull request by kartikeys98**
fixes #4753

**For short term contributors:** we understand that getting your commits well
defined like we require is a hard task and takes some learning. If you
look to help without wanting to contribute long term there's no need
for you to learn this. Just drop us a message and we'll take care of brushing
up your stuff for merge!

### Checklist

- [x] I read the [commit guidelines](http://coala.io/commit) and I've followed
      them.
- [x] I ran coala over my code locally. (*All commits have to pass
      individually.* It is not sufficient to have "fixup commits" on your PR,
      our bot will still report the issues for the previous commit.) You will
      likely receive a lot of bot comments and build failures if coala does not
      pass on every single commit!

After you submit your pull request, **DO NOT click the 'Update Branch' button.**
When asked for a rebase, consult [coala.io/rebase](https://coala.io/rebase)
instead.

Please consider helping us by reviewing other peoples pull requests as well:

- pick up any PR at <https://coala.io/review>
- review it (check <https://coala.io/reviewing> for more info)
- if you are sure that it needs work, use `cobot mark wip <URL>` to get it out
  of the review queue.

The more you review, the more your score will grow at coala.io and we will
review your PRs faster!

### CODE REVIEW

**Comment by kartikeys98** [comment]
Does it look fine now? :)

**Comment by prnvdixit** [comment]
ack de275b9d1ca297f987b21ed4c0a8ce21162f1080

**Comment by prnvdixit** [comment]
unack de275b9d1ca297f987b21ed4c0a8ce21162f1080

**Comment by kartikeys98** [comment]
@prnvdixit Done.

**Comment by kartikeys98** [comment]
@prnvdixit Okay, now I suppose it renders correctly.

**Comment by prnvdixit** [comment]
ack a04dae2ad30c5d1f820e643ad647469fbb17e881
Good work @kartikeys98 👍 😄.

**Comment by prnvdixit** [comment]
ack a04dae2ad30c5d1f820e643ad647469fbb17e881

File: docs/Developers/Writing_Native_Bears.rst [hunk 1]
```diff
@@ -20,6 +20,11 @@ directory of the coala-tutorial repository.
     If you want to wrap an already existing tool, please refer to
     :doc:`this tutorial instead<Writing_Linter_Bears>`.
 
+.. note::
+
+    Also refer `Python3 regex guide <https://docs.python.org/3/library/re.html>`_ for understanding regex.
+    For further information refer `this <http://www.diveintopython3.net/regular-expressions.html>`_.
```
Comment (Makman2) [inline docs/Developers/Writing_Native_Bears.rst]: Imo this should be placed into writing-linter-bears under the regex-output-format section. This is kinda unrelated here.

File: docs/Developers/Writing_Native_Bears.rst [hunk 2]
```diff
@@ -20,6 +20,11 @@ directory of the coala-tutorial repository.
     If you want to wrap an already existing tool, please refer to
     :doc:`this tutorial instead<Writing_Linter_Bears>`.
 
+.. note::
+
+    Also refer `Python3 regex guide <https://docs.python.org/3/library/re.html>`_ for understanding regex.
```
Comment (Makman2) [inline docs/Developers/Writing_Native_Bears.rst]: --> refer **to**

Comment (Makman2) [inline docs/Developers/Writing_Native_Bears.rst]: for ~~understanding~~ more information on ~~regex~~ regular expressions.

Comment (Makman2) [reply docs/Developers/Writing_Native_Bears.rst]: You should also concatenate the following sentence with this one by just writing "<linkA> and <linkB>"

File: docs/Developers/Writing_Native_Bears.rst (hunk 1 above)
Comment (Makman2) [inline docs/Developers/Writing_Native_Bears.rst]: Could you rephrase "this" and be more precise? Like "refer to this tutorial" or so (if it's one)

**Comment by Makman2** [comment]
unack a04dae2

**Comment by kartikeys98** [comment]
Please review. I've updated it accordingly.

**Comment by kartikeys98** [comment]
Does it look right?

File: docs/Developers/Writing_Linter_Bears.rst [hunk 3]
```diff
@@ -145,8 +145,10 @@ that is printed out.
 For the exact list of named groups ``@linter`` recognizes, see the `API
 documentation <https://api.coala.io/en/latest/>`__.
 
-For more info generally on regexes, see `Python re module
-<https://docs.python.org/3/library/re.html>`_.
+.. note::
+
+    Please refer to `Python3 re module <https://docs.python.org/3/library/re.html>`_ for understanding regular expression.
```
Comment (prnvdixit) [inline docs/Developers/Writing_Linter_Bears.rst]: *expression* -> expression**s** :)

File: docs/Developers/Writing_Linter_Bears.rst [hunk 4]
```diff
@@ -145,8 +145,10 @@ that is printed out.
...
+    For further information refer `diveintopython <http://www.diveintopython3.net/regular-expressions.html>`_'s regex tutorial.
```
Comment (prnvdixit) [inline docs/Developers/Writing_Linter_Bears.rst]: *diveintopython*'s regex tutorial -> *Dive into Python's chapter on regexes*
(Italics are hyperlinked)

**Comment by kartikeys98** [comment]
Concatenated the 2 links.

**Comment by kartikeys98** [comment]
Hmm, why did the test failed on circleci. :thinking:

**Comment by soumya-ranjan7** [comment]
ack 8f3693c

File: docs/Developers/Writing_Linter_Bears.rst [hunk 5]
```diff
@@ -145,8 +145,9 @@ that is printed out.
 For the exact list of named groups ``@linter`` recognizes, see the `API
 documentation <https://api.coala.io/en/latest/>`__.
 
-For more info generally on regexes, see `Python re module
-<https://docs.python.org/3/library/re.html>`_.
+.. note::
+
+    Please refer to `Python3 re module <https://docs.python.org/3/library/re.html>`_ for understanding regular expressions and also refer `Dive into python <http://www.diveintopython3.net/regular-expressions.html>`_'s chapter on regexes.
```
Comment (Makman2) [inline docs/Developers/Writing_Linter_Bears.rst]: 1. please break the line, it's too long ;)
2. --> refer **to** (for the second occurrence)
3. I think you can compact the sentence to:
```
Please refer to `Python3 re module <...>` and `Dive into python <...>` for information about regular expressions.
```

**Comment by Makman2** [comment]
Also commit message:
~~c~~Closes (upper-case C 😄)

**Comment by Makman2** [comment]
unack 8f3693c

**Comment by kartikeys98** [comment]
Done the required.

File: docs/Developers/Writing_Linter_Bears.rst [hunk 6]
```diff
@@ -145,8 +145,9 @@ that is printed out.
 For the exact list of named groups ``@linter`` recognizes, see the `API
 documentation <https://api.coala.io/en/latest/>`__.
 
-For more info generally on regexes, see `Python re module
-<https://docs.python.org/3/library/re.html>`_.
+.. note::
+
+    Please refer to `Python3 re module <https://docs.python.org/3/library/re.html>`_ and `Dive into python <http://www.diveintopython3.net/regular-expressions.html>`_'s for information about regular expressions.
```
Comment (Makman2) [inline docs/Developers/Writing_Linter_Bears.rst]: please still break the line 😅 
Maximum 80 chars per line if possible :)

Comment (Makman2) [reply docs/Developers/Writing_Linter_Bears.rst]: Also: You can remove the `'s` behind the second link^^

File: docs/Developers/Writing_Linter_Bears.rst [hunk 7]
```diff
@@ -145,8 +145,10 @@ that is printed out.
 For the exact list of named groups ``@linter`` recognizes, see the `API
 documentation <https://api.coala.io/en/latest/>`__.
 
-For more info generally on regexes, see `Python re module
-<https://docs.python.org/3/library/re.html>`_.
+.. note::
+
+    Please refer to `Python3 re module <https://docs.python.org/3/library/re.html>`_ and `Dive into python <http://www.diveintopython3.net/regular-expressions.html>`_'s for information
```
Comment (prnvdixit) [inline docs/Developers/Writing_Linter_Bears.rst]: This doesn't fits right 😄
> Dive into python's for information

Comment (kartikeys98) [reply docs/Developers/Writing_Linter_Bears.rst]: Okay well I've tried different methods for breaking it into new line but it ignores it. Any ideas?

Comment (prnvdixit) [reply docs/Developers/Writing_Linter_Bears.rst]: I meant grammatically actually 😄 - What happens if you break the line at ```and``` (Remember, no trailing spaces)

File: docs/Developers/Writing_Linter_Bears.rst [hunk 8]
```diff
@@ -145,8 +145,10 @@ that is printed out.
 For the exact list of named groups ``@linter`` recognizes, see the `API
 documentation <https://api.coala.io/en/latest/>`__.
 
-For more info generally on regexes, see `Python re module
-<https://docs.python.org/3/library/re.html>`_.
+.. note::
+
+    | Please refer to `Python3 re module <https://docs.python.org/3/library/re.html>`_ and `Dive into python <http://www.diveintopython3.net/regular-expressions.html>`_'s for information
```
Comment (prnvdixit) [inline docs/Developers/Writing_Linter_Bears.rst]: Umm.. You don't need to break the line inside ```actual doc``` - Do it inside ```code``` only, as it makes reading code easier on all devices 😃

Comment (kartikeys98) [reply docs/Developers/Writing_Linter_Bears.rst]: I'm sorry I don't understand which code.

**Comment by kartikeys98** [comment]
@prnvdixit Please review and see if it fits in right.

File: docs/Developers/Writing_Linter_Bears.rst [hunk 9]
```diff
@@ -145,8 +145,10 @@ that is printed out.
 For the exact list of named groups ``@linter`` recognizes, see the `API
 documentation <https://api.coala.io/en/latest/>`__.
 
-For more info generally on regexes, see `Python re module
-<https://docs.python.org/3/library/re.html>`_.
+.. note::
```
Comment (prnvdixit) [inline docs/Developers/Writing_Linter_Bears.rst]: Remove this ```note::``` - Just write it with original style & indentation as was in
```
For more info generally on regexes, see `Python re module
<https://docs.python.org/3/library/re.html>`_.
```
Break the lines at appropriate points so that no line exceeds 80 chars.. 😄

Comment (kartikeys98) [reply docs/Developers/Writing_Linter_Bears.rst]: Done.

File: docs/Developers/Writing_Linter_Bears.rst [hunk 10]
```diff
@@ -145,8 +145,8 @@ that is printed out.
 For the exact list of named groups ``@linter`` recognizes, see the `API
 documentation <https://api.coala.io/en/latest/>`__.
 
-For more info generally on regexes, see `Python re module
-<https://docs.python.org/3/library/re.html>`_.
+Please refer to `Python3 re module <https://docs.python.org/3/library/re.html>`_ and
```
Comment (prnvdixit) [inline docs/Developers/Writing_Linter_Bears.rst]: 80 chars please 😄

File: docs/Developers/Writing_Linter_Bears.rst [hunk 11]
```diff
@@ -145,8 +145,8 @@ that is printed out.
...
+`Dive into python <http://www.diveintopython3.net/regular-expressions.html>`_ for information about regular expressions.
```
Comment (prnvdixit) [inline docs/Developers/Writing_Linter_Bears.rst]: Same here - You can use another line actually 😉

**Comment by kartikeys98** [comment]
ack c6758cc

**Comment by Makman2** [comment]
ack c6758cc

**Comment by Makman2** [comment]
@gitmate-bot rebase

**Comment by sils** [comment]
@gitmate-bot rebase

**Comment by kartikeys98** [comment]
Why did some checks fail?

**Comment by sils** [comment]
we've had GitMate.io deployment issues and are working on resolving the issue for future cases. Right now GitMate is operational again, just do an empty amend and force push to retrigger the CI pipeline.

**Comment by kartikeys98** [comment]
Pushed again.

**Comment by kartikeys98** [comment]
Removed

**Comment by sils** [comment]
ack https://github.com/coala/coala/pull/4760/commits/9a5085711c29d5b359569d0b94f50652cbfdae0f

**Comment by sils** [comment]
@rultor merge

**Comment by prnvdixit** [comment]
@kartikeys98 Good work... 👍 😄

**Comment by kartikeys98** [comment]
Thanks @prnvdixit for being calm and guiding me for my first contribution.
